### PR TITLE
Add New Extension WeChat

### PR DESCRIPTION
**Pull request by RaffeYang**
## Description

# WeChat

A [Raycast](https://www.raycast.com) extension for WeChat to quickly search your contacts and chat.

cover

showcase

## Requirement

1. Install [WeChat for Mac](https://www.wechat.com) 
2. Install [WeChatTweak-macOS](https://github.com/Sunnyyoung/WeChatTweak-macOS) 

## Todo

- [x] Search Contact
- [x] List Search Contact Numbers
- [x] Open Contact
- [x] List WeChat ID
- [x] Copy WeChat ID
- [x] Display Avatar
- [x] Fix some bug
- [x] Upload to Raycast Store
- [x] Detect if Wechat is installed

## How to Use

before I create the final version and release it to the Raycast Store,
you can use it with:

`npm install && npm run dev`

## License

[MIT](LICENSE)

## Screencast

https://user-images.githubusercontent.com/7557357/211519985-5725b2a4-fb86-4e7c-8211-62b997cf51ca.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/wechat/src/util/isWeChatTweakInstalledAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Alert, confirmAlert, open, closeMainWindow } from "@raycast/api"
+
+export const isWeChatTweakInstalledAlertDialog = async () => {
+  const options: Alert.Options = {
+    icon: { source: "wechat.png" },
+    title: "WeChatTweak is Not Installed",
```
Comment (pernielsentikaer) [inline extensions/wechat/src/util/isWeChatTweakInstalledAlert.tsx:6]:
```suggestion
    title: "WeChatTweak is not installed",
```

File: extensions/wechat/src/util/isWeChatInstalledAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Alert, confirmAlert, open, closeMainWindow } from "@raycast/api"
+
+export const isWeChatInstalledAlertDialog = async () => {
+  const options: Alert.Options = {
+    icon: { source: "wechat.png" },
+    title: "WeChat is Not Installed",
```
Comment (pernielsentikaer) [inline extensions/wechat/src/util/isWeChatInstalledAlert.tsx:6]:
```suggestion
    title: "WeChat is not installed",
```

File: extensions/wechat/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  "author": "raffeyang",
+  "categories": [
+    "Communication"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "WeChat",
+      "subtitle": "Search your WeChat contacts and chat",
```
Comment (pernielsentikaer) [inline extensions/wechat/package.json:16]:
```suggestion
      "title": "Search Contacts and Chat",
      "subtitle": "WeChat",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* cover.png should not be added as a metadata image
* video.mp4 should not be added as a metadata image

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/wechat/src/util/isWeChatTweakInstalledAlert.tsx (hunk 1 above)
Comment (raffeyang) [reply extensions/wechat/src/util/isWeChatTweakInstalledAlert.tsx:6]: Nice suggestion.

**Comment by raffeyang** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * cover.png should not be added as a metadata image
> * video.mp4 should not be added as a metadata image
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I have followed your suggestion to update the code, thanks!

**Comment by pernielsentikaer** [comment]
Can we host the mp4 elsewhere, so we don't have to commit that large file? You can just use the RAW link from you own repo fork if it is 😊

**Comment by raffeyang** [comment]
> Can we host the mp4 elsewhere, so we don't have to commit that large file? You can use the RAW link from your own repo fork if it is 😊

Solved,review again.

**Pull request merged**